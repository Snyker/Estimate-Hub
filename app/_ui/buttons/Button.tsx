import React, {ButtonHTMLAttributes, PropsWithChildren} from "react";

export const Button: React.FC<PropsWithChildren<ButtonHTMLAttributes<HTMLButtonElement>>> = ({ children, className = '', ...props}) => {
    return (
        <button type="button" className={`btn btn-primary ${className}`} {...props}>{children}</button>
    );
};
