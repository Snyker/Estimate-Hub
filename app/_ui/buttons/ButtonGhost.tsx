import React, {ButtonHTMLAttributes, PropsWithChildren} from "react";

export const ButtonGhost: React.FC<PropsWithChildren<ButtonHTMLAttributes<HTMLButtonElement>>> = ({ children, className= '', ...props}) => {
    return (
        <button className={`btn btn-ghost ${className}`} type={'button'} {...props}>
            {children}
        </button>
    );
};
