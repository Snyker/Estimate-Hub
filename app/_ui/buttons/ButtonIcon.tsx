import React, {ButtonHTMLAttributes, PropsWithChildren} from "react";

export const ButtonIcon: React.FC<PropsWithChildren<ButtonIconProps>> = ({ children, className='', icon = (<></>), side = 'left', ...props}) => {
    return (
        <button type="button" className={`btn btn-rounded p-2 w-8 h-8 gap-2 ${className} ${side === 'left' ? 'flex-row' : 'flex-row-reverse'}`} {...props}>
            {icon && icon}
            {children}
        </button>
    );
};

export type ButtonIconProps = ButtonHTMLAttributes<HTMLButtonElement> & {
    side?: 'left' | 'right'
    icon?: React.ReactElement
}
