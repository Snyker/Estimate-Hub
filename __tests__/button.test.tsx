import React from 'react';
import renderer from 'react-test-renderer';
import {Button} from "@/app/_ui/buttons/Button";
import {ButtonIcon, ButtonIconProps} from "@/app/_ui/buttons/ButtonIcon";

describe('Button Component', () => {
    it('renders correctly with children', () => {
        const tree = renderer.create(<Button>Click me</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies custom class', () => {
        const tree = renderer.create(<Button className="custom-class">Click me</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handles onClick event', () => {
        const mockOnClick = jest.fn();
        const tree = renderer.create(<Button onClick={mockOnClick}>Click me</Button>);

        // Simuler un clic
        tree.root.findByType('button').props.onClick();

        expect(mockOnClick).toHaveBeenCalled();
    });
});

describe('ButtonIcon Component', () => {
    const defaultProps: ButtonIconProps = {
        children: 'Click me',
    };

    it('renders correctly with default props', () => {
        const tree = renderer.create(<ButtonIcon {...defaultProps} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies custom class', () => {
        const tree = renderer.create(<ButtonIcon {...defaultProps} className="custom-class" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies icon on the left by default', () => {
        const tree = renderer.create(<ButtonIcon {...defaultProps} icon={<span>ğŸš€</span>} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies icon on the right when side is set to "right"', () => {
        const tree = renderer.create(<ButtonIcon {...defaultProps} icon={<span>ğŸš€</span>} side="right" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handles onClick event', () => {
        const mockOnClick = jest.fn();
        const tree = renderer.create(<ButtonIcon {...defaultProps} onClick={mockOnClick} />);

        // Simuler un clic
        tree.root.findByType('button').props.onClick();

        expect(mockOnClick).toHaveBeenCalled();
    });
});

