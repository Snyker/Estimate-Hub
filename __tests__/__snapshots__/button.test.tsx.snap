// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component applies custom class 1`] = `
<button
  className="btn btn-primary custom-class"
  type="button"
>
  Click me
</button>
`;

exports[`Button Component renders correctly with children 1`] = `
<button
  className="btn btn-primary "
  type="button"
>
  Click me
</button>
`;

exports[`ButtonIcon Component applies custom class 1`] = `
<button
  className="btn gap-2 custom-class flex-row"
  type="button"
>
  Click me
</button>
`;

exports[`ButtonIcon Component applies icon on the left by default 1`] = `
<button
  className="btn gap-2  flex-row"
  type="button"
>
  <span>
    ðŸš€
  </span>
  Click me
</button>
`;

exports[`ButtonIcon Component applies icon on the right when side is set to "right" 1`] = `
<button
  className="btn gap-2  flex-row-reverse"
  type="button"
>
  <span>
    ðŸš€
  </span>
  Click me
</button>
`;

exports[`ButtonIcon Component renders correctly with default props 1`] = `
<button
  className="btn gap-2  flex-row"
  type="button"
>
  Click me
</button>
`;
